source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -481,10 +697,10 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
         // Start traversal from this output
         root = output;
     }
-    else if (element->isA<Node>())
-    {
-        NodePtr node = element->asA<Node>();
 
+    NodePtr node = element->asA<Node>();
+    if (node && (node->getType() != "surfaceshader"))
```

Comment:
Seems unnecessary to convert to node and check for node type if the isA<Output>() branch above has already been taken. I think this should be changed back to an `else if (element->isA<Node>())` statement. And the type check for "surfaceshader" or not can be done inside that branch.